stingray/simulator/events.py

```diff
@@ -174,3 +177,48 @@ def gen_lc_from_events(event_list, bin_time, start_time=None,
         times = times + bin_time / 2.
 
     return times, lc.astype(np.float)
+
+def assign_energies(event_list, spectrum):
+
+    '''
+    Assign energies to event lists.
+
+    Parameters
+    ----------
+    event_list: array-like
+        Times of arrival of events
+    spectrum: 2-d array
+        Energies versus corresponding fluxes
+
+    Returns
+    -------
+    assigned_energies: array-like
+        Energies assigned to all events in event list
+    '''
+
+    assert spectrum.shape[0] != 2, "Spectrum must be a 2-d array."
+
+    assert np.all(np.isfinite(event_list)), "There are inf or NaN values in " \
+                                          "your event list!"
+
+    # Cast inputs as numpy arrays
+    event_list = np.array(event_list)
```

Comment:
With the current setup, the event_list array doesn't look useful. Why not just giving an integer number of photons to generate as input?